### ISSUE TITLE
sc.get_added_obs_importance: reset_zero_weight strange behavior

### ISSUE THREAD

**Issue by lauraschachter** [pypest/pyemu#286]
Expected behavior is that if reset_zero_weight isn't specified, the zero-weighted observations are reset to weight = 1.0, but if this is non-numeric, then weights are not reset. In order to get this to work, the user must set reset_zero_weight = 1.0. Either the docstring should be rewritten to be accurate, or the behavior of the code should change to match the docstring.

From the docstring:
```
    reset_zero_weight (`bool`, optional)
                a flag to reset observations with zero weight in either
                `obslist_dict` or `base_obslist`. If `reset_zero_weights`
                passed as a `float`,then that value will be assigned to
                zero weight obs.  Otherwise, zero-weight obs will be given a
                weight of 1.0.  Default is `False`.
```

From the code : (note that the weights are not reset to 1.0 like mentioned above)
```reset = False
   if reset_zero_weight is not False:
            if not self.obscov.isdiagonal:
                raise NotImplementedError(
                    "cannot reset weights for non-" + "diagonal obscov"
                )
            reset = True
            try:
                weight = float(reset_zero_weight)
            except:
                weight = 1.0
            self.logger.statement("resetting zero weights to {0}".format(weight))
            # make copies of the original obscov and pst
            # org_obscov = self.obscov.get(self.obscov.row_names)
            org_obscov = self.obscov.copy()
            org_pst = self.pst.get()
```

**Comment by mnfienen** [comment]
It seems like the code snippet @lauraschachter showed is doing two things. It's setting the reweighting weight to 1.0 by default, but only if `reset_zero_weight` is not `False`. It's also checking for a diagonal obscov since reweighting only makes sense if obscov is diagonal.

I propose adding a variable called `added_obs_weight` which is a float. If it is `None`, the default will be 1.0. Then, a separate variable `reset_zero_weight` which is truly boolean. If it is `True`, do the check for diagonal obscov and `raise` if obscov is not diagonal and also use `added_obs_weight` value for resetting, or default to 1.0. 

If it is `False`, do not reset weights at all (which only makes sense if obscov has off-diagonal terms, right?).

**Comment by mnfienen** [comment]
OK, after some discussion offline, we have a solution. There is a bit of a conceptual change here.

Previously, `reset_zero_weights` applied to both `obslist_dict` which is the new, potential observations _and_ `base_obslist` which is the user-identified calibration set. However, users should be providing only non-zero weighted observations in `base_obslist`. Also, the user should be able to provide custom weights for their potential observations and have no resetting happen. Finally, a common workflow might have all potential obs supplied with zero weight and the user may want just a saingle new weight applied to them all. This result is a bit of a mashup of the above logic and the previous implementation. 

To summarize:

- `reset_zero_weights` will now only apply to `obslist_dict` and `pyemu` will throw an exception if `base_obslist` includes and obs with zero weight
- `reset_zero_weights` will be a true `float` now (`False` will not be an allowable value).
         -  If `reset_zero_weights == 0` no adjustments will be made to weights. 
         -  If `reset_zero_weights > 0` weights will be set to the float value in `reset_zero_weights`
         - Default value will be `1.0`
         - Docstring will be updated to reflect this behavior

**Comment by jtwhite79** [comment]
agree with that!  So now the default is the generic behavior most users will be expecting: passing a dict of zero-weighted obs for testing and those obs should be assigned a nominal weight automatically under the hood.

**Comment by mnfienen** [comment]
One question is, do we still need this exception? 
```
            if not self.obscov.isdiagonal:
                raise NotImplementedError(
                    "cannot reset weights for non-" + "diagonal obscov"
                )
```

I ask because, what if the `obscov` for the calibration data is not diagonal, but for potential obs it is diagonal?
